Fixes to unsafe chapter (#177)
google/comprehensive-rust: src/unsafe/raw-pointers.md

    let r1 = &mut num as *mut i32;
    let r2 = &num as *const i32;

>     // Safe because r1 and r2 were obtained from references and so are guaranteed to be non-null and

Reviewer: Now that I think about this again: is it not already UB to read from the immutable `r2` when we have the mutable `r1` existing at the same time? That is, I think I've seen that Rust demands that you uphold the normal borrow checker rules, even when you use `unsafe`. I'm not very sure of this, though.

Reviewer: Okay, cool!

Reviewer: Please consider taking out the comment inside the `unsafe` block — the comment here covers it. Also consider making the comment here shorter. Perhaps just

```suggestion
    // Safe because r1 and r2 are aligned and live pointers.
```

Author: I've removed the comment inside the block. I do think it's important that the safety comment here explains exactly how we know that the requirements are met, rather than just parroting what they are. Writing it out in full forces people to think through exactly what they are doing and why.
